File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/resources/webapp/js/ds/FiscalPeriodLovDs.js
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */

Ext.define("net.nan21.dnet.module.md.base.period.ds.model.FiscalPeriodLovDs", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"code", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"year", type:"string"},
		{name:"yearId", type:"int", useNull:true}
	]});

Ext.define("net.nan21.dnet.module.md.base.period.ds.model.FiscalPeriodLovDsFilter", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean", useNull:true},
		{name:"code", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"year", type:"string"},
		{name:"yearId", type:"int", useNull:true},
		{name:"yearId_From",type:"int", useNull:true},
		{name:"yearId_To",type:"int", useNull:true}
	]
});

File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/base/period/ds/model/FiscalPeriodLovDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.base.period.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeWithCodeLov;
import net.nan21.dnet.module.md.base.period.domain.entity.FiscalPeriod;

@Ds(entity = FiscalPeriod.class, jpqlWhere = " e.active = true ", sort = {@SortField(field = FiscalPeriodLovDs.f_code)})
public class FiscalPeriodLovDs extends AbstractTypeWithCodeLov<FiscalPeriod> {

	public static final String f_yearId = "yearId";
	public static final String f_year = "year";

	@DsField(join = "left", path = "year.id")
	private Long yearId;

	@DsField(join = "left", path = "year.code")
	private String year;

	public FiscalPeriodLovDs() {
		super();
	}

	public FiscalPeriodLovDs(FiscalPeriod e) {
		super(e);
	}

	public Long getYearId() {
		return this.yearId;
	}

	public void setYearId(Long yearId) {
		this.yearId = yearId;
	}

	public String getYear() {
		return this.year;
	}

	public void setYear(String year) {
		this.year = year;
	}
}
